### PR TITLE
[DOCS] Reformats validate API

### PR DESCRIPTION
**Pull request by szabosteve**
Relates to elastic/docs#937 and https://github.com/elastic/elasticsearch/issues/45621.

This PR updates the validate API to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Depends on: https://github.com/elastic/elasticsearch/pull/46377/ (it adds `query` to the common params list.)

Resources:
* [Validate API spec](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/indices.validate_query.json)
* [Validate API doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-validate.html)

### CODE REVIEW

File: docs/reference/search/validate.asciidoc [hunk 1]
```diff
@@ -1,8 +1,74 @@
...
 
-The validate API allows a user to validate a potentially expensive query
-without executing it. We'll use the following test data to explain _validate:
+Validates a potentially expensive query without executing it.
+
+[source,console]
+--------------------------------------------------
+GET twitter/_validate/query?q=user:foo
+--------------------------------------------------
+
```
Comment (jrodewig) [inline docs/reference/search/validate.asciidoc]:
```suggestion
// TEST[setup:twitter]

```

File: docs/reference/search/validate.asciidoc [hunk 2]
```diff
@@ -14,13 +80,15 @@ PUT twitter/_bulk?refresh
 --------------------------------------------------
 // TESTSETUP
```
Comment (jrodewig) [inline docs/reference/search/validate.asciidoc]:
```suggestion
```
This is causing the CI to fail. `// TESTSETUP` must be the first comment in a file.

File: docs/reference/search/validate.asciidoc [hunk 3]
```diff
@@ -14,13 +80,15 @@ PUT twitter/_bulk?refresh
...
 
+
 When sent a valid query:
 
 [source,console]
 --------------------------------------------------
 GET twitter/_validate/query?q=user:foo
 --------------------------------------------------
```
Comment (jrodewig) [inline docs/reference/search/validate.asciidoc]:
```suggestion
// TEST[continued]

```

File: docs/reference/search/validate.asciidoc [hunk 4]
```diff
@@ -78,9 +122,9 @@ GET twitter/_validate/query
 NOTE: The query being sent in the body must be nested in a `query` key, same as
```
Comment (jrodewig) [inline docs/reference/search/validate.asciidoc]: The prior snippet needs a `// TEST[continued]` comment.

File: docs/reference/search/validate.asciidoc [hunk 5]
```diff
@@ -101,8 +145,10 @@ GET twitter/_validate/query
 --------------------------------------------------
 // TESTRESPONSE
```
Comment (jrodewig) [inline docs/reference/search/validate.asciidoc]: The prior `GET twitter/_validate/query` snippet needs a `// TEST[continued]` comment.

File: docs/reference/search/validate.asciidoc [hunk 6]
```diff
@@ -117,7 +163,8 @@ GET twitter/_validate/query?explain=true
 }
 --------------------------------------------------
 
-responds with:
+
```
Comment (jrodewig) [inline docs/reference/search/validate.asciidoc]: The prior snippet needs a `// TEST[continued]` comment.

File: docs/reference/search/validate.asciidoc [hunk 7]
```diff
@@ -181,14 +229,15 @@ Response:
...
+===== Rewrite and all_shards parameters
+
 By default, the request is executed on a single shard only, which is randomly
 selected. The detailed explanation of the query may depend on which shard is
 being hit, and therefore may vary from one request to another. So, in case of
 query rewrite the `all_shards` parameter should be used to get response from
 all available shards.
 
-For Fuzzy Queries:
-
```
Comment (jrodewig) [inline docs/reference/search/validate.asciidoc]: The following snippet requires this hidden setup:

```
////
[source,console]
--------------------------------------------------
PUT twitter/_bulk?refresh
{"index":{"_id":1}}
{"user" : "kimchy", "post_date" : "2009-11-15T14:12:12", "message" : "trying out Elasticsearch"}
{"index":{"_id":2}}
{"user" : "kimchi", "post_date" : "2009-11-15T14:12:13", "message" : "My username is similar to @kimchy!"}
--------------------------------------------------
////
```

File: docs/reference/search/validate.asciidoc [hunk 8]
```diff
@@ -1,8 +1,74 @@
...
+without executing it. The query can be sent either as a path parameter or in the 
+request body.
+
+
+[[search-validate-api-path-params]]
+==== {api-path-parms-title}
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=index]
+
+`query`::
```
Comment (jrodewig) [inline docs/reference/search/validate.asciidoc]: This should probably be moved to `common-parms.asciidoc`.

**Review by jrodewig — CHANGES_REQUESTED** [review]
Left a few comments that should fix the CI. Thanks!

File: docs/reference/search/validate.asciidoc [hunk 9]
```diff
@@ -78,9 +123,9 @@ GET twitter/_validate/query
 NOTE: The query being sent in the body must be nested in a `query` key, same as
```
Comment (szabosteve) [inline docs/reference/search/validate.asciidoc]:
```suggestion
// TEST[continued]
NOTE: The query being sent in the body must be nested in a `query` key, same as
```

File: docs/reference/search/validate.asciidoc (hunk 8 above)
Comment (szabosteve) [reply docs/reference/search/validate.asciidoc]: https://github.com/elastic/elasticsearch/pull/46377/ adds the parameter.

File: docs/reference/search/validate.asciidoc [hunk 10]
```diff
@@ -202,7 +263,8 @@ GET twitter/_validate/query?rewrite=true&all_shards=true
 }
 --------------------------------------------------
 
-Response:
+
```
Comment (jrodewig) [inline docs/reference/search/validate.asciidoc]: The prior snippet needs `// TEST[continued]`.

Comment (szabosteve) [reply docs/reference/search/validate.asciidoc]: @jrodewig Thank you so much, it solved the problem!

**Review by jrodewig — APPROVED** [review]
LGTM

**Comment by szabosteve** [comment]
Backported to 7.x, 7.4, and 7.3.

**Pull request merged**